tests/functional/conftest.py

```diff
@@ -0,0 +1,27 @@
+import os
+import subprocess
+from time import sleep
+
+import pytest
+
+
+@pytest.fixture(scope="session", autouse=True)
+def install_dcgm_snap():
+    """Install the snap and enable dcgm-exporter service for testing."""
+    snap = os.environ["TEST_SNAP"]
+    dcgm_exporter_service = "snap.dcgm.dcgm-exporter.service"
+    assert 0 == subprocess.run(["sudo", "snap", "install", "--dangerous", snap]).returncode
+
+    subprocess.run(["sudo", "systemctl", "enable", "--now", dcgm_exporter_service])
```

Comment:
You should be able to use ``snap start` or not ?

Reply:
Agreed; we're testing the snap - can you use the `snap` subcommands (services, start, stop, restart) instead of systemctl?